### DISCUSSION TITLE
VS Code extension that compiles locally

### DISCUSSION THREAD

**Discussion by verhovsky** [compiler-explorer/compiler-explorer#5956] [Ideas]
I want a local version of Compiler Explorer in my code editor (VS Code). I want to "explore" just one compiler for just one language (my locally installed C compiler, clang or gcc, for my computer's architecture) and not upload my code somewhere. Mostly out of curiosity.

It should

- run each time my file changes and the file is parsable. I think it shouldn't throw out old compilations like Compiler Explorer says "`<Compilation failed>`" but something less distracting like just greying out the code from the previous run until the file is compilable. When I save the file with Cmd+S it can show the compilation error message but while I'm typing it shouldn't distract me
- highlight the lines in the assembly that correspond to the line of C code my cursor is on and scroll the lines as I move the cursor (this might be difficult to keep my place when there's a complex correspondence between C and Assembly and I've scrolled the Assembly window, you don't want to make me lose my place by jumping the Assembly window back to the first matching instruction every time otherwise if there's a particular part I'm interested in I would have to manually scroll back every time there's a change)
- be fast (ideally it would re-use compilation/parsing work from the previous run of things that haven't changed but I don't think clang or gcc can do that)
- not take up much screen width (because I'm focusing on my C code, not on the assembly)
- not need any configuration. If I've already configured VS Code enough to run/debug my code with F5, that should be enough for the extension as well. If I just create a new file, set its language to C and type in a hello world without saving the file or anything, it should just work. If it allowed further customization that might be useful, but it shouldn't be required
- allow hovering over assembly instructions for details like Compiler Explorer does as well

It could support other languages that Compiler Explorer does as well.

Anyway, hope that someone reads this and is inspired to make an MVP (or it already exists?) because I'm not gonna.

**Comment by partouf** [comment]
I believe this extension doesn't transmit anything to us, but just uses some of the asm parsing that we wrote: https://marketplace.visualstudio.com/items?itemName=harikrishnan94.cxx-compiler-explorer
(source: https://github.com/harikrishnan94/cxx-compiler-explorer)

Not sure how they compile and if they use the current project or something
